### PR TITLE
document release process

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://github.com/openshift/jenkins-plugin/issues/99

@bparees FYI

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+git rebase upstream/master
+git push origin master
+git push origin --tags
+```
+
+## Submit the new release to the Jenkins organization
+
+1. Assumptions: your Git ID has push access to the two repositories for this plugin; your Jenkins ID (https://wiki.jenkins-ci.org/display/JENKINS/User+Account+on+Jenkins) is listed in https://github.com/jenkins-infra/repository-permissions-updater/blob/master/permissions/plugin-openshift-pipeline.yml
+1. Then run `mvn release:prepare release:perform`
+	- NOTE:  You'll minimally be prompted for the `release version`, `release tag`, and the `new development version` (typically a version indicator that ends with `-SNAPSHOT`).  Also, if we happen to need to depend on a SNAPSHOT version of openshift-restclient-java, you'll get prompted about moving off the SNAPSHOT version, but override this.  Stay on the SNAPSHOT version of openshift-restclient-java if recent updates to that dependency that we need have not yet been cut to a new FINAL release.
```
Comment (bparees) [inline RELEASING.md]: the instructions about snapshot vs final, when/if you'll get prompted, and what you should/shouldn't do if you do get prompted, are a bit confusing.

also some guidance about what values to use for release version, release tag, and new development version would be helpful.

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+1. Run `git push https://github.com/openshift/jenkins-plugin.git master` to upload the 2 commits created for cutting the new release to our upstream, development repository, and get the two repositories back in sync.
+1. Monitor https://updates.jenkins-ci.org/download/plugins/openshift-pipeline/ for the existance of the new version of the plugin.  Warning: the link for the new version will show up, but does not mean the `openshift-pipeline.hpi` file is available yet.  Click the link to confirm you can download the new version of the plugin.  When you can download the latest `openshift-pipeline.hpi` file, the process is complete, and the new release is available.
```
Comment (bparees) [inline RELEASING.md]: s/existance/existence/

Comment (gabemontero) [reply RELEASING.md]: just pushed correction

File: RELEASING.md (hunk 1 above)
Comment (gabemontero) [reply RELEASING.md]: OK I've attempted to clarify / clear up.  Let me know if it makes better sense to humans besides me :-)

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+1. Then run `mvn release:prepare release:perform`
+	- NOTE:  You'll minimally be prompted for the `release version`, `release tag`, and the `new development version`.  Default choices will be provided for each, and the defaults are typically acceptable, so you can just hit the enter key for all three prompts.  As an example, if we are currently at v1.0.36, it will provide 1.0.37 for the new `release version` and `release tag`.  For the `new development version` it will provide 1.0.38-SNAPSHOT, which is again acceptable.  The only time you *might* have to override the defaul provided is if we happen to need to depend on a SNAPSHOT version of openshift-restclient-java (i.e. `5.3.0-SNAPSHOT`).  This occurs when we add new features to openshift-restclient-java, but the eclipse team has not cut a new, official release (which will typically look like `5.3.0-FINAL`).  If we are in such a mode, you'll get prompted about moving off the SNAPSHOT version (the default provided would be `5.3.0`), but override this (i.e. type in `5.3.0-SNAPSHOT`).
```
Comment (bparees) [inline RELEASING.md]: s/defaul/default/

Comment (bparees) [reply RELEASING.md]: s/i.e./e.g./  ("that is to say" vs "for example")

Comment (bparees) [reply RELEASING.md]: s/if we happen to need to depend on/if we currently depend on/

Comment (bparees) [reply RELEASING.md]: (the i.e.->e.g. comment only applies to the first instance, the second usage is correct, or at least can be correct either way in my head)

**Comment by bparees** [comment]
couple more suggestions and lgtm.

File: RELEASING.md (hunk 3 above)
Comment (gabemontero) [reply RELEASING.md]: Updates incorporated ... about to push

**Comment by gabemontero** [comment]
Updates pushed ... also some markdown lang cleanup based on the markdown preview tab I found in eclipse

**Pull request merged**